---
title: "Explanation"
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{Explanation}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(thecentraldogma)
```

# Packages description

## random_dna

**Description:** The random_dna function creates a random string of DNA
of nucleic bases (A,T,G,C) with a user defined lenght.

**Parameters:**

**lenght_of_DNA:** must be an integer. Defines the lenght of the DNA
string.

Examples:

```{r}
random_dna(lenght_of_dna = 10)
#Outputs a string of dna bases that's 10 long.
```

## replacement_of_matches

Description:

-   Takes a DNA sequence and replaces all "T" with "U" resulting in the
    corresponding RNA sequence.

Parameters:

-   DNA_sequence: must be a string

Function:

-   gsub: replacement of matches

Usage:

```         
replace_T_with_U(DNA_sequence)
```

Examples:

```{r}
RNA_sequence <- replace_T_with_U(DNA_sequence = "GATTA")
print(RNA_sequence)
```

## seq_to_codons

**Description**

seq_to_codons is a function that takes a nucleotide sequence and splits it into the codons, starting at the position specified.

**Usage**

```         
seq_to_codons(sequence, start = 1)
```

**Arguments**

|            |                                                                                      |
|-------------|-----------------------------------------------------------|
|------------|--------------------------------------------------------------------------------------|
| `sequence` | A nucleotide sequence                                                                |
| `start`    | The first position in the nucleotide sequence that should start the codon splitting. |

The codons of the input sequence is outputted as a vector of strings.

**Examples**

```{r}
seq_to_codons("ACTGATCATGA", start=1)
```

## translate_codons

**Description\
**The function translate_codons translates a vector containing
nucleotide triplet(s) (codons) to a single string of amino acids.

**Parameters**\
"codons" is a vector containing triplet(s) of nucleotides.\
"standard_codon_table" is a predefined table containing the standard
codons and their corresponding amino acid.

**Example usage**

```{r}
codons <- c("AUG", "UUU", "UGA") 
translate_codons(codons) 
```

## plot_count

This function plots the frequency of amino acids.

The parameter *input_string* is a character string that is analyzed for
unique character frequencies.

The function uses the packages stringr to split a string then counts
frequency of character. Lastly it plots the frequency in columns.

**Example:**

```{r}
plot <- amino_frequency_plot_local("SYRKHDAPNA")
print(plot)
```
